### PR TITLE
feat: Apps-3068 SEO friendly SectionPagination

### PR DESCRIPTION
**Pull request by farosFreed**
Connected to [APPS-3068](https://jira.library.ucla.edu/browse/APPS-3068)

**Component Updated:** SectionPagination.vue

**Stories Updated:** ~/stories/SectionPagination.stories.js

**Notes:**

- changed pagination buttons to be elements with href tags. To do this we added a `generateLink` method which computes the right href path
- also added a prop generateLinkCallback which allows parent components to override the default generateLink function

I was not able to find an SEO evaluation tool that would notice the buttons did / did not have href tags (they only notice <a> tags without hrefs), so I have simply verified in the inspect window that the links have href tags now

**Checklist:**

-   [X] I checked that it is working locally in the dev server
-   [X] I checked that it is working locally in the storybook
-   [X] I checked that it is working locally in the 
ftva dev server
-   [X] I added a screenshot of it working
-   ~~[ ] UX has reviewed and approved this~~
-   [ ] I assigned this PR to someone on the dev team to review
-   [X] I used a conventional commit message
-   [X] I assigned myself to this PR

[APPS-3068]: https://uclalibrary.atlassian.net/browse/APPS-3068?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by farosFreed** [comment]
Screenshot 2024-11-26 at 3 59 58 PM

**Review by pghorpade — APPROVED** [review]
Love the flexibility of passing function

File: src/stories/SectionPagination.stories.js [hunk 1]
```diff
@@ -42,10 +40,12 @@ export function LastPage() {
   }
 }
 
+// this story uses the generateLinkCallback prop
+// to generate the links in the library-website-nuxt format instead of the default format
 export function WithPagesAndCurrentPage() {
   return {
     components: { SectionPagination },
-    template: '<section-pagination :pages="23" :initialCurrentPage="4" />',
+    template: '<section-pagination :pages="23" :initialCurrentPage="4" :generateLinkCallback="(x) => { return `/page/` + x }"/>',
```
Comment (pghorpade) [inline src/stories/SectionPagination.stories.js:48]: i think this should be return `?page=` + x?

Comment (pghorpade) [reply src/stories/SectionPagination.stories.js:48]: shouldnt it be query param?

**Review by pghorpade — CHANGES_REQUESTED** [review]

File: src/stories/SectionPagination.stories.js (hunk 1 above)
Comment (farosFreed) [reply src/stories/SectionPagination.stories.js:48]: I based the pattern on the way the data looks for the default stories ( /page/1, /page/10). But I will also include a story that uses the queryparam in the path to make sure that works without any more tinkering. I see the live implementation on the library site looks like 'https://www.library.ucla.edu/search-site?q=new&from=20' so I'll try to implement that

Comment (farosFreed) [reply src/stories/SectionPagination.stories.js]: I tweaked the story to use the callback to send the link in the same format as the library site

**Review by pghorpade — APPROVED** [review]
thank you 👍

**Pull request merged**